js/src/contracts/badgereg.js

```diff
@@ -18,40 +18,61 @@ import { bytesToHex, hex2Ascii } from '~/api/util/format';
 
 import ABI from './abi/certifier.json';
 
-const ZERO = '0x0000000000000000000000000000000000000000000000000000000000000000';
+const ZERO20 = '0x0000000000000000000000000000000000000000';
+const ZERO32 = '0x0000000000000000000000000000000000000000000000000000000000000000';
 
 export default class BadgeReg {
   constructor (api, registry) {
     this._api = api;
     this._registry = registry;
 
     registry.getContract('badgereg');
-    this.certifiers = {}; // by name
+    this.certifiers = []; // by id
     this.contracts = {}; // by name
   }
 
-  fetchCertifier (name) {
-    if (this.certifiers[name]) {
-      return Promise.resolve(this.certifiers[name]);
+  nrOfCertifiers () {
+    return this._registry.getContract('badgereg')
+      .then((badgeReg) => {
+        return badgeReg.instance.badgeCount.call({}, [])
+          .then((count) => count.valueOf());
+      });
+  }
+
+  fetchCertifier (id) {
+    if (this.certifiers[id]) {
+      return Promise.resolve(this.certifiers[id]);
     }
     return this._registry.getContract('badgereg')
       .then((badgeReg) => {
-        return badgeReg.instance.fromName.call({}, [name])
-        .then(([ id, address ]) => {
-          return Promise.all([
-            badgeReg.instance.meta.call({}, [id, 'TITLE']),
-            badgeReg.instance.meta.call({}, [id, 'IMG'])
-          ])
-            .then(([ title, img ]) => {
-              title = bytesToHex(title);
-              title = title === ZERO ? null : hex2Ascii(title);
-              if (bytesToHex(img) === ZERO) img = null;
+        return badgeReg.instance.badge.call({}, [ id ]);
+      })
+      .then(([ address, name ]) => {
+        if (address === ZERO20) throw new Error(`Certifier ${id} does not exist.`);
```

Comment:
Considder putting this across 2 lines with a blank space between the blocks -

1. Readability, not gaining anything (I know beauty is in the eye of the beerholder)
2. When we add the tests and have coverage information, the line will show up as covered even when just the condition is evaluated - when it is in a proper block, the info will show that the error is actually tested for or now. (Goes for everything, even ?:).